tracing-subscriber/src/fmt/format.rs

```diff
@@ -280,8 +309,54 @@ where
     }
 }
 
-// === impl FormatFields ===
+#[cfg(feature = "json")]
+impl<N, T> FormatEvent<N> for Format<Json, T>
+where
+    N: for<'writer> FormatFields<'writer>,
+    T: FormatTime,
+{
+    fn format_event(
+        &self,
+        ctx: &span::Context<'_, N>,
+        writer: &mut dyn fmt::Write,
+        event: &Event<'_>,
+    ) -> fmt::Result {
+        let mut timestamp = String::new();
+        self.timer.format_time(&mut timestamp)?;
+
+        #[cfg(feature = "tracing-log")]
+        let normalized_meta = event.normalized_metadata();
+        #[cfg(feature = "tracing-log")]
+        let meta = normalized_meta.as_ref().unwrap_or_else(|| event.metadata());
+        #[cfg(not(feature = "tracing-log"))]
+        let meta = event.metadata();
+
+        let mut visit = || {
+            let mut serializer = Serializer::new(WriteAdaptor::new(writer));
+            let mut serializer = serializer.serialize_map(None)?;
+
+            serializer.serialize_entry("timestamp", &timestamp)?;
+            serializer.serialize_entry("level", &meta.level().as_serde())?;
```

Comment:
do you think the source code location (file name and line number) from the event's metadata should be serialized as well? this could, perhaps, be configurable...

Reply:
I think we have this already from visiting the event. At least when I run this code with our server, the output has the line numbers etc. available. It's not configurable though I guess...

Reply:
Looking at the expected output in your test, I don't see the source code location: https://github.com/tokio-rs/tracing/blob/223cbed844e0f939649da979ff635800008828f4/tracing-subscriber/src/fmt/format.rs#L819

The source location is part of the event's metadata, _not_ the event's fields --- the visitor won't see it.

Reply:
Seems to happen only when using `tracing-log`. Here's an example from prisma server:

```json
{"timestamp":"Oct 14 14:00:27.687","level":"INFO","target":"prisma","message":"foo","log.target":"prisma","log.module_path":"prisma","log.file":"query-engine/prisma/src/main.rs","log.line":153}
```

Reply:
Ahh, okay — the `log.$name` fields are specific to `tracing-log`, and will only occur on `tracing` events converted from `log` records. 

Those are included because metadata from `log` records cannot easily be used in `tracing` metadata due to lifetime mismatches; the `NormalizeMetadata` trait bridges the gap here a bit. These fields should probably be skipped if `NormalizeMetadata` is being used, and we should instead serialize the `tracing` metadata so that the source locations can be recorded for `tracing` events as well as `log` records...